src/OneOff.php

```diff
@@ -30,4 +31,30 @@ public function isExpired(DateTimeInterface $dateTime): bool
     {
         return $this->dateTime < $dateTime;
     }
+
+    public function getStart(): ?DateTimeInterface
+    {
+        return $this->dateTime;
+    }
+
+    public function getEnd(): ?DateTimeInterface
+    {
+        return $this->getStart();
+    }
+
+    public static function fromJson(string $data): Frequency
+    {
+        $data = json_decode($data, true);
+        if (empty($data)) {
+            throw new InvalidArgumentException('Can\'t create one-off frequency without datetime');
+        }
+
+        return new static(new DateTime($data));
+    }
+
+    #[\ReturnTypeWillChange]
+    public function jsonSerialize()
+    {
+        return json_encode($this->dateTime->format('Ymd\THis'));
```

Comment:
Maybe use a standard format here? (One of `DateTimeInterface::`'s like `DateTimeInterface::ATOM`?) Yeah, it's internal anyway, but a custom one isn't of any advantage anyway.

Reply:
I made use of this format because the internal rrule library is also using this format.